integrations/opsgenie/main.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any
+from loguru import logger
+import asyncio
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from port_ocean.context.ocean import ocean
+from client import OpsGenieClient
+from utils import ObjectKind
+
+CONCURRENT_REQUESTS = 5
+
+
+def init_client() -> OpsGenieClient:
+    return OpsGenieClient(
+        ocean.integration_config["api_token"],
+        ocean.integration_config.get("api_url", "https://api.opsgenie.com"),
+    )
+
+
+async def enrich_services_with_team_data(
+    opsgenie_client: OpsGenieClient,
+    semaphore: asyncio.Semaphore,
+    service: dict[str, Any],
+) -> dict[str, Any]:
+    async with semaphore:
+        team_data = await opsgenie_client.get_oncall_team(service["teamId"])
+        schedule = await opsgenie_client.get_schedule_by_team(service["teamId"])
+        related_incidents = await opsgenie_client.get_incidents_by_service(
+            service["id"]
+        )
```

Comment:
Gather them together
```suggestion
team_data, schedule, related_incidents = asyncio.gather(
opsgenie_client.get_oncall_team(service["teamId"]),
opsgenie_client.get_schedule_by_team(service["teamId"]),
opsgenie_client.get_incidents_by_service(service["id"])
)
```